core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -783,6 +790,8 @@ abstract class RDD[T: ClassTag](
   @deprecated("use mapPartitionsWithIndex and filter", "1.0.0")
   def filterWith[A](constructA: Int => A)(p: (T, A) => Boolean): RDD[T] = withScope {
     mapPartitionsWithIndex((index, iter) => {
+      /* val cleanP = sc.clean(p)
+      val cleanA = sc.clean(constructA) */
```

Comment:
@ted-yu I investigated this a little and this won't work because you're bringing in the whole `SparkContext` into the closure. Instead we need to clean the closures outside of `mapPartitionsWithIndex` as you have done in other methods. There is nothing special about `filterWith`.